src/NServiceBus.Core/Transports/OutgoingMessage.cs

```diff
@@ -0,0 +1,32 @@
+namespace NServiceBus.Transports
+{
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// The message going out to the transport
+    /// </summary>
+    public class OutgoingMessage
+    {
+        /// <summary>
+        /// Constructs the message
+        /// </summary>
+        /// <param name="headers">The headers associated with this message</param>
+        /// <param name="body">The body of the message</param>
+        public OutgoingMessage(Dictionary<string, string> headers,byte[] body)
```

Comment:
After doing some spiking with @danielmarbach (and @simoncropp ) we failed
to come up with a api using streams that makes sense for out going messages.

We're not saying that it's a bad idea, just that we couldn't figure it out.
Happy for someone else trying to take it on.

Action points for me:

* Adjust the incoming message to expose a `StreamReader` instead of a
stream since only reading is allowed
* Start a separate pull that aims to make sure we use the reader all the
way up to the deserializer (right now we copy to a byte[] right away)

On Wed, Mar 11, 2015 at 9:30 AM, John Simons <notifications@github.com>
wrote:

> In src/NServiceBus.Core/Transports/OutgoingMessage.cs
> <https://github.com/Particular/NServiceBus/pull/2678#discussion_r26195046>
> :
>
> > @@ -0,0 +1,32 @@
> > +namespace NServiceBus.Transports
> > +{
> > +    using System.Collections.Generic;
> > +
> > +    /// <summary>
> > +    /// The message going out to the transport
> > +    /// </summary>
> > +    public class OutgoingMessage
> > +    {
> > +        /// <summary>
> > +        /// Constructs the message
> > +        /// </summary>
> > +        /// <param name="headers">The headers associated with this message</param>
> > +        /// <param name="body">The body of the message</param>
> > +        public OutgoingMessage(Dictionary<string, string> headers,byte[] body)
>
> Rabbit supports stream too
> … <#14c07f546effe994_>
> On Wednesday, March 11, 2015, Andreas Öhlund <notifications@github.com>
> wrote: In src/NServiceBus.Core/Transports/OutgoingMessage.cs <
> https://github.com/Particular/NServiceBus/pull/2678#discussion_r26193063>
> : > @@ -0,0 +1,32 @@ > +namespace NServiceBus.Transports > +{ > + using
> System.Collections.Generic; > + > + /// <summary> > + /// The message going
> out to the transport > + /// </summary> > + public class OutgoingMessage >
> + { > + /// <summary> > + /// Constructs the message > + /// </summary> > +
> /// <param name="headers">The headers associated with this message</param>
> > + /// <param name="body">The body of the message</param> > + public
> OutgoingMessage(Dictionary<string, string> headers,byte[] body) Oh, I
> didn't think about that. I guess that makes sense then. Since all
> transports except Rabbit support a stream in some way I guess use exposing
> a stream is the most future proof way to do it? … <#14c07c1374fd4d21_> On
> Tue, Mar 10, 2015 at 10:28 PM, Simon Cropp <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote: In
> src/NServiceBus.Core/Transports/OutgoingMessage.cs <
> https://github.com/Particular/NServiceBus/pull/2678#discussion_r26168170>
> : > @@ -0,0 +1,32 @@ > +namespace NServiceBus.Transports > +{ > + using
> System.Collections.Generic; > + > + /// <summary> > + /// The message going
> out to the transport > + /// </summary> > + public class OutgoingMessage >
> + { > + /// <summary> > + /// Constructs the message > + /// </summary> > +
> /// <param name="headers">The headers associated with this message</param>
> > + /// <param name="body">The body of the message</param> > + public
> OutgoingMessage(Dictionary<string, string> headers,byte[] body) This is
> outgoing messages so it will already be in memory since the users will have
> "new'ed" up the message? yes so you will already have allocated all the
> memory no. since the act of serializing the message to a byte array
> effectively takes a, larger, copy of the message into memory. effectively
> taking up the memory space twice. — Reply to this email directly or view it
> on GitHub <
> https://github.com/Particular/NServiceBus/pull/2678/files#r26168170>. —
> Reply to this email directly or view it on GitHub <
> https://github.com/Particular/NServiceBus/pull/2678/files#r26193063>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/2678/files#r26195046>.
>